tools/Voron.Recovery/Recovery.cs

```diff
@@ -910,11 +962,14 @@ private bool ValidateOverflowPage(PageHeader* pageHeader, byte* eof, long startO
 
                 if (checksum != pageHeader->Checksum)
                 {
+                    CheckInvalidPagesInARaw(pageHeader, mem);
                     var message =
                         $"Invalid checksum for overflow page {pageHeader->PageNumber}, expected hash to be {pageHeader->Checksum} but was {checksum}";
                     mem = PrintErrorAndAdvanceMem(message, mem);
                     return false;
                 }
+
+                _InvalidChecksumWithNoneZeroMac = 0;
```

Comment:
If we were able to get a _single_ valid page, we don't need to deal with this any longer.